packages/devtools_app/lib/src/timeline/flutter/timeline_flame_chart.dart

```diff
@@ -561,23 +574,45 @@ class AsyncGuidelinePainter extends CustomPainter {
   ) {
     for (int i = firstLineIndex; i < guidelines.length; i++) {
       final line = guidelines[i];
+      // Take [chartStartInset] and
+      // [FullTimelineFlameChart.asyncGuidelineOffset] into account when
+      // calculating [zoomedLine] because these units of space should not scale.
+      final unzoomableOffset =
+          FullTimelineFlameChart.asyncGuidelineOffset + chartStartInset;
+
+      LineSegment zoomedLine;
+      final zoomedLineStartX =
+          (line.start.dx - unzoomableOffset) * zoom + unzoomableOffset;
+      if (line is VerticalLineSegment) {
```

Comment:
rather than using is checks, prefer to use virtualMethods.
For example: add a `line.toZoomed(zoom: zoom. offset: unzoomableOffset)` method